### ISSUE TITLE
Cannot find mpv-1.dll

### ISSUE THREAD

**Issue by Nearata** [jaseg/python-mpv#104]
**Operating System**: Windows 10 Home (1909)
**Python Version**: 3.8
**MPV Version**: 0.45

### Description
Cannot find mpv-1.dll but the file is in the same directory of the script.

### Folder structure
```
- vvvvid_downloader
-- mpv.py
-- mpv-1.dll
-- test_mpv.py
```

### Code

```
# test_mpv.py

url = "https://example.com/vod.m3u8"
player = MPV(osc=True)
player.play(url)
player.wait_for_playback()
player.terminate()
```

### Traceback
```
Traceback (most recent call last):
  File "test_mpv.py", line 2, in <module>
    from mpv import MPV
  File "E:\GitHub Workspace\vvvvid-downloader\vvvvid_downloader\mpv.py", line 33, in <module>
    raise OSError('Cannot find mpv-1.dll in your system %PATH%. One way to deal with this is to ship mpv-1.dll '
OSError: Cannot find mpv-1.dll in your system %PATH%. One way to deal with this is to ship mpv-1.dll with your script and put the directory your script is in into %PATH% before "import mpv": os.environ["PATH"] = os.path.dirname(__file__) + os.pathsep + os.environ["PATH"] If mpv-1.dll is located elsewhere, you can add that path to os.environ["PATH"].
```

**Comment by McSinyx** [comment]
Have you tried doing what is suggested by the error?

**Comment by Nearata** [comment]
@McSinyx Yes but it worked before the lastest update.

However, i fixed like this:

```
# mpv.py

os.add_dll_directory(os.getcwd())
if os.name == 'nt':
    # dll = ctypes.util.find_library('mpv-1.dll')
    dll = CDLL("mpv-1.dll")
    ...
    # backend = CDLL(dll)
    backend = dll
    fs_enc = 'utf-8'
```

**Comment by McSinyx** [comment]
> it worked before the lastest update

That was not very secured so it was fixed sometime ago.  See #96 for discussion.

**Comment by ghsanti** [comment]
for some reason mine was not downloaded from the mpv installation with scoop (new to windows here.)

I basically downloaded it from https://sourceforge.net/projects/mpv-player-windows/files/libmpv/

extracted the `libmpv.dll` file where my dir is and it worked provided that `pip install yt-dlp` as well.

**Comment by jaseg** [comment]
I have updated the DLL search code to look next to mpv.py, and I have improved on the error messages, which I hope will help with what was mentioned in this issue. These improvements will be in the next release.

On Windows, mpv.py now searches for mpv.dll everywhere in `%PATH%`, and if it cannot find it there, it will also search the directory `mpv.py` is in. If nothing is found, you now get the error message from above. Right now, accepted file names are `mpv-2.dll`, `libmpv-2.dll`, and `mpv-1.dll` in that order.

Side note: Relative paths in `%PATH%` will most likely not work.

Thanks y'all for the reports and comments.
